fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/NestingKeyExpression.java

```diff
@@ -201,4 +201,14 @@ public int planHash() {
     public boolean equalsAtomic(AtomKeyExpression other) {
         return this.getClass() == other.getClass() && parent.equals(((NestingKeyExpression) other).parent);
     }
+
+    @Override
+    public boolean equivalentForSort(@Nonnull KeyExpression other) {
+        if (getClass() != other.getClass()) {
+            return false;
+        }
+
+        NestingKeyExpression that = (NestingKeyExpression)other;
+        return this.parent.equals(that.parent) && this.getChild().equivalentForSort(that.getChild());
```

Comment:
I noticed this is using `.equals` on `parent`. I think that doesn't matter here because the `NullStandIn` information shouldn't really make a difference on the parents of nested expressions, but should it be `equivalentForSort`?

Reply:
Yes, that was deliberate. The parent in this case does not affect the ordering. If there were different possible interpretations of the parent for something else, and that should be ignored here, then we'd maybe need an `equivalentForParentOfSort`?

Reply:
Hm, I suppose if we were dotting all i's and crossing all t's, technically, because the `NullStandIn` value of the parent doesn't matter at all here, I guess `equivalentForParentOfSort` would ignore the `NullStandIn` information entirely. Adding a new one of those for all expressions seems like overkill. Adding one just on `FieldKeyExpression`...maybe makes sense?

The argument on the other side would be that setting the null interpretation on the parent of a nested field doesn't make any sense, so maybe we just ignore that case.

I don't think I have particularly strong feelings here, one way or the other. I'm also happy to take this as `.equals`.

Reply:
I added a short comment that this is intentional.